main.go

```diff
@@ -15,7 +15,7 @@ type Album  struct {
   ID      int64
   Title   string
   Artist  string
-  Price   float
+  Price   float32
```

Comment:
In real financial applications, representing any form of money with a `float`-like type is a **very bad** idea, as floating point numbers do *not* have finite precision.

The mathematical reason why is that there are countably-infinite (1, 2, 3, 4, ...) many 32-bit numbers (indeed it's 2^32 of them, about 4 billion), but there are uncountably-infinite many real numbers. If the idea of "uncountably-infinite" is new to you, consider the question "what's the 'next' number after 0.123?". If you think about it, this makes no sense. It might be tempting to say 0.124 but 0.1231 is closer. And 0.1230000001 is even closer than that.

As a result, if we need to represent an infinitely precise number (1/3, π, etc.) in a computer system we have to approximate, and the approximation is bad enough that it leads to inaccurate calculations, which can have disastrous implications / result in lawsuits.

The solution in most cases is to change a price in e.g. in dollars (say $12.56) to cents (1256) and represent it as an integer in code. Of course, this too is limited, in the sense that there's a max and a min integer value, but we can typically know upfront how large a price will be. Contrast that to floating point numbers which could have "holes" of inaccuracies at much smaller numbers.

Reply:
Will be addressed after working on unit testing.